File: Scala/app/models/Product.scala
package models

import play.api.libs.json._

case class Product(id: Int, name: String, description: String, price: Double, stock: Int)

object Product {
  implicit val productReads: Reads[Product] = Json.reads[Product]
  implicit val productWrites: Writes[Product] = Json.writes[Product]
}

File: Go/models/product.go
package models

import "gorm.io/gorm"

type Product struct {
	gorm.Model
	Name        string   `json:"name"`
	Description string   `json:"description"`
	Price       float32  `json:"price"`
	Stock       int      `json:"stock"`
	CategoryID  uint     `json:"category"`
	Category    Category `gorm:"foreignKey:CategoryID"`
}
